### ISSUE TITLE
clang-tidy doesn't report missing forward when one parameter is forwarded, but some other parameter isn't

### ISSUE THREAD

**Issue by geza-herman** [llvm/llvm-project#83845]
In this example, `foo()` doesn't forward `y`, but clang-tidy doesn't report it.

If none of the parameters are forwarded (comment the `use(std::forward<X>(x));` line), then `cppcoreguidelines-missing-std-forward` correctly reports both parameters as non-forwarded. But if one of them is forwarded, then the other one is not reported.

```c++
#include <utility>

template <typename X>
void use(const X &x) {}

template <typename X, typename Y>
void foo(X &&x, Y &&y) {
    use(std::forward<X>(x));
    use(y);
}
```

**Comment by jcsxky** [comment]
We should remove `optionally` to ensure the function call uses correct argument.
https://github.com/llvm/llvm-project/blob/0c4736338596d6e527e286b7b551af4bb8b63a55/clang-tools-extra/clang-tidy/cppcoreguidelines/MissingStdForwardCheck.cpp#L117
But,  this would break
https://github.com/llvm/llvm-project/blob/0c4736338596d6e527e286b7b551af4bb8b63a55/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/missing-std-forward.cpp#L170-L173
